internal/distributed/proxy/httpserver/utils.go

```diff
@@ -1679,3 +1680,170 @@ func RequestHandlerFunc(c *gin.Context) {
 	}
 	c.Next()
 }
+
+func generateTemplateArrayData(list []interface{}) *schemapb.TemplateArrayValue {
+	dtype := getTemplateArrayType(list)
+	var data *schemapb.TemplateArrayValue
+	switch dtype {
+	case schemapb.DataType_Bool:
+		result := make([]bool, len(list))
+		for i, item := range list {
+			result[i] = item.(bool)
+		}
+		data = &schemapb.TemplateArrayValue{
+			Data: &schemapb.TemplateArrayValue_BoolData{
+				BoolData: &schemapb.BoolArray{
+					Data: result,
+				},
+			},
+		}
+	case schemapb.DataType_String:
+		result := make([]string, len(list))
+		for i, item := range list {
+			result[i] = item.(string)
+		}
+		data = &schemapb.TemplateArrayValue{
+			Data: &schemapb.TemplateArrayValue_StringData{
+				StringData: &schemapb.StringArray{
+					Data: result,
+				},
+			},
+		}
+	case schemapb.DataType_Int64:
+		result := make([]int64, len(list))
+		for i, item := range list {
+			result[i] = int64(item.(float64))
+		}
+		data = &schemapb.TemplateArrayValue{
+			Data: &schemapb.TemplateArrayValue_LongData{
+				LongData: &schemapb.LongArray{
+					Data: result,
+				},
+			},
+		}
+	case schemapb.DataType_Float:
+		result := make([]float64, len(list))
+		for i, item := range list {
+			result[i] = item.(float64)
+		}
+		data = &schemapb.TemplateArrayValue{
+			Data: &schemapb.TemplateArrayValue_DoubleData{
+				DoubleData: &schemapb.DoubleArray{
+					Data: result,
+				},
+			},
+		}
+	case schemapb.DataType_Array:
+		result := make([]*schemapb.TemplateArrayValue, len(list))
+		for i, item := range list {
+			result[i] = generateTemplateArrayData(item.([]interface{}))
+		}
+		data = &schemapb.TemplateArrayValue{
+			Data: &schemapb.TemplateArrayValue_ArrayData{
+				ArrayData: &schemapb.TemplateArrayValueArray{
+					Data: result,
+				},
+			},
+		}
+	case schemapb.DataType_JSON:
+		result := make([][]byte, len(list))
+		for i, item := range list {
+			bytes, err := json.Marshal(item)
+			// won't happen
+			if err != nil {
+				panic(fmt.Sprintf("marshal data(%v) fail, please check it!", item))
+			}
+			result[i] = bytes
+		}
+		data = &schemapb.TemplateArrayValue{
+			Data: &schemapb.TemplateArrayValue_JsonData{
+				JsonData: &schemapb.JSONArray{
+					Data: result,
+				},
+			},
+		}
+	// won't happen
+	default:
+		panic(fmt.Sprintf("Unexpected data(%v) type when generateTemplateArrayData, please check it!", list))
+	}
+	return data
+}
+
+func getTemplateArrayType(value []interface{}) schemapb.DataType {
+	dtype := getTemplateType(value[0])
+
+	for _, v := range value {
+		if getTemplateType(v) != dtype {
+			return schemapb.DataType_JSON
+		}
+	}
+	return dtype
+}
+
+func getTemplateType(value interface{}) schemapb.DataType {
+	switch v := value.(type) {
+	case bool:
+		return schemapb.DataType_Bool
+	case string:
+		return schemapb.DataType_String
```

Comment:
add int64? If it's an int number like 1, 2, 3, will asserting it as int64 succeed?

Reply:
no, all the number in json can only be asserted as float64.